Mathlib/RingTheory/HahnSeries/Summable.lean

```diff
@@ -427,6 +444,40 @@ theorem hsum_sub {R : Type*} [Ring R] {s t : SummableFamily Γ R α} :
     (s - t).hsum = s.hsum - t.hsum := by
   rw [← lsum_apply, LinearMap.map_sub, lsum_apply, lsum_apply]
 
+theorem isPWO_iUnion_support_prod_mul {s : α → HahnSeries Γ R} {t : β → HahnSeries Γ R}
+    (hs : (⋃ a, (s a).support).IsPWO) (ht : (⋃ b, (t b).support).IsPWO) :
+    (⋃ (a : α × β), ((fun a ↦ ((s a.1) * (t a.2))) a).support).IsPWO :=
+  isPWO_iUnion_support_prod_smul hs ht
+
+theorem finite_co_support_prod_mul (s : SummableFamily Γ R α)
+    (t : SummableFamily Γ R β) (g : Γ) :
+    Finite {(a : α × β) | ((fun (a : α × β) ↦ (s a.1 * t a.2)) a).coeff g ≠ 0} :=
+  finite_co_support_prod_smul s t g
+
+/-- A summable family given by pointwise multiplication of a pair of summable families. -/
+@[simps]
+def FamilyMul (s : SummableFamily Γ R α) (t : SummableFamily Γ R β) :
+    (SummableFamily Γ R (α × β)) where
+  toFun a := s (a.1) * t (a.2)
+  isPWO_iUnion_support' :=
+    isPWO_iUnion_support_prod_mul s.isPWO_iUnion_support t.isPWO_iUnion_support
+  finite_co_support' g := finite_co_support_prod_mul s t g
+
+theorem familymul_eq_familysmul {β : Type*} (s : SummableFamily Γ R α) (t : SummableFamily Γ R β) :
+    FamilyMul s t = FamilySMul s t :=
```

Comment:
Hmm, I see we've maybe already gone wrong with names... Not too late to fix! :-)